clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminApiHandler.java

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin.internals;
+
+import org.apache.kafka.common.requests.AbstractRequest;
+import org.apache.kafka.common.requests.AbstractResponse;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public interface AdminApiHandler<K, V> {
+
+    /**
+     * Get a user-friendly name for the API this handler is implementing.
+     */
+    String apiName();
+
+    /**
+     * Initialize the set of keys required to handle this API and how the driver
+     * should map them to the broker that will handle the request for these keys.
+     *
+     * Two mapping types are supported:
+     *
+     * - Static mapping: when the brokerId is known ahead of time
+     * - Dynamic mapping: when the brokerId must be discovered dynamically
+     *
+     * @return the key mappings
+     */
+    Keys<K> initializeKeys();
+
+    /**
+     * Build the request. The set of keys are derived by {@link AdminApiDriver}
+     * during the lookup stage as the set of keys which all map to the same
+     * destination broker.
+     *
+     * @param brokerId the target brokerId for the request
+     * @param keys the set of keys that should be handled by this request
+     *
+     * @return a builder for the request containing the given keys
+     */
+    AbstractRequest.Builder<?> buildRequest(Integer brokerId, Set<K> keys);
+
+    /**
+     * Callback that is invoked when a request returns successfully.
+     * The handler should parse the response, check for errors, and return a
+     * result which indicates which keys (if any) have either been completed or
+     * failed with an unrecoverable error.
+     *
+     * It is also possible that the response indicates an incorrect target brokerId
+     * (e.g. in the case of a NotLeader error when the request is bound for a partition
+     * leader). In this case the key will be "unmapped" from the target brokerId
+     * and lookup will be retried.
+     *
+     * Note that keys which received a retriable error should be left out of the
+     * result. They will be retried automatically.
+     *
+     * @param brokerId the brokerId that the associated request was sent to
+     * @param keys the set of keys from the associated request
+     * @param response the response received from the broker
+     *
+     * @return result indicating key completion, failure, and unmapping
+     */
+    ApiResult<K, V> handleResponse(Integer brokerId, Set<K> keys, AbstractResponse response);
+
+    class Keys<K> {
+        public final Map<K, Integer> staticKeys;
+        public final Set<K> dynamicKeys;
+        public final AdminApiLookupStrategy<K> lookupStrategy;
+
+        public Keys(
```

Comment:
Is it legal that key exists in both `staticKeys` and `dynamicKeys`? If not, could you add a check for them?